### ISSUE TITLE
IntelliSense not getting include directories from external targets/interfaces

### ISSUE THREAD

**Issue by longnguyen2004** [microsoft/vscode-cmake-tools#1429]
### Brief Issue Summary

I'm working on a project that uses SFML as a dependency. However, after adding the exported targets into my project, I still get IntelliSense errors on SFML headers. Then I found out that the include directories aren't added into IntelliSense search dir list

### Expected:

CMake Tools gets include directories from external targets and append it to IntelliSense search dir list

### Apparent Behavior:

Include directories aren't added into search dir list

### CMake Tools Log

```
[main] Configuring folder: Mania Replay Renderer 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_TOOLCHAIN_FILE:FILEPATH=C:\llvm-mingw\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET:STRING=x64-windows "-Hg:/Code/CMake based projects/Mania Replay Renderer" "-Bg:/Code/CMake based projects/Mania Replay Renderer/build" -G "Visual Studio 16 2019" -T host=x64 -A x64
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19041.
[cmake] -- The C compiler identification is MSVC 19.27.29111.0
[cmake] -- The CXX compiler identification is MSVC 19.27.29111.0
[cmake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
[cmake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe - works
[cmake] -- Detecting C compiler ABI info
[cmake] -- Detecting C compiler ABI info - done
[cmake] -- Detecting C compile features
[cmake] -- Detecting C compile features - done
[cmake] -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
[cmake] -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe - works
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- Found SFML 2.5.1 in C:/llvm-mingw/vcpkg/installed/x64-windows/share/sfml
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: G:/Code/CMake based projects/Mania Replay Renderer/build
```

### Developer Tools Log

```
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.205Z [debug] [main] Detected saving of CMakeLists.txt, attempting automatic reconfigure...
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.208Z [info] [main] Configuring folder: Mania Replay Renderer 
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.209Z [debug] [main] Saving open files before configure/build
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.902Z [debug] [driver] Run _refreshExpansions
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.903Z [debug] [driver] Run _refreshExpansions cb
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.906Z [debug] [driver] Start configure 
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.906Z [debug] [driver] Runnnig pre-configure checks and steps
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.907Z [debug] [driver] Using CMake toolchain Visual Studio Build Tools 2019 - amd64 with vcpkg for configuring
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.919Z [debug] [driver] Run _refreshExpansions
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.919Z [debug] [driver] Run _refreshExpansions cb
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.960Z [debug] [cmakefileapi-driver] Invoking CMake C:\Program Files\CMake\bin\cmake.EXE with arguments ["--no-warn-unused-cli","-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE","-DCMAKE_TOOLCHAIN_FILE:FILEPATH=C:\\llvm-mingw\\vcpkg\\scripts\\buildsystems\\vcpkg.cmake","-DVCPKG_TARGET_TRIPLET:STRING=x64-windows","-Hg:/Code/CMake based projects/Mania Replay Renderer","-Bg:/Code/CMake based projects/Mania Replay Renderer/build","-G","Visual Studio 16 2019","-T","host=x64","-A","x64"]
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:20.973Z [info] [proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_TOOLCHAIN_FILE:FILEPATH=C:\llvm-mingw\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET:STRING=x64-windows "-Hg:/Code/CMake based projects/Mania Replay Renderer" "-Bg:/Code/CMake based projects/Mania Replay Renderer/build" -G "Visual Studio 16 2019" -T host=x64 -A x64
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:21.303Z [info] [cmake] Not searching for unused variables given on the command line.
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:22.788Z [info] [cmake] -- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19041.
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:46.378Z [info] [cmake] -- The C compiler identification is MSVC 19.27.29111.0
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:52.111Z [info] [cmake] -- The CXX compiler identification is MSVC 19.27.29111.0
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:31:52.749Z [info] [cmake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:06.547Z [info] [cmake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe - works
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:06.638Z [info] [cmake] -- Detecting C compiler ABI info
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:16.186Z [info] [cmake] -- Detecting C compiler ABI info - done
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:16.220Z [info] [cmake] -- Detecting C compile features
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:16.232Z [info] [cmake] -- Detecting C compile features - done
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:16.321Z [info] [cmake] -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:24.915Z [info] [cmake] -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe - works
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:24.934Z [info] [cmake] -- Detecting CXX compiler ABI info
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:33.758Z [info] [cmake] -- Detecting CXX compiler ABI info - done
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:33.769Z [info] [cmake] -- Detecting CXX compile features
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:33.775Z [info] [cmake] -- Detecting CXX compile features - done
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:34.020Z [info] [cmake] -- Found SFML 2.5.1 in C:/llvm-mingw/vcpkg/installed/x64-windows/share/sfml
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:34.022Z [info] [cmake] -- Configuring done
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:34.201Z [info] [cmake] -- Generating done
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:34.544Z [info] [cmake] -- Build files have been written to: G:/Code/CMake based projects/Mania Replay Renderer/build
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:34.594Z [debug] [cmakefileapi-parser] Read reply folder: g:\Code\CMake based projects\Mania Replay Renderer\build\.cmake\api\v1\reply
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:34.596Z [debug] [cmakefileapi-parser] Found index files: ["cache-v2-4a525591e110fa85e667.json","codemodel-v2-79ff303cf98a9817673b.json","index-2020-09-10T11-32-34-0542.json","target-ALL_BUILD-Debug-76821e378ebd7f961b28.json","target-ALL_BUILD-MinSizeRel-76821e378ebd7f961b28.json","target-ALL_BUILD-Release-76821e378ebd7f961b28.json","target-ALL_BUILD-RelWithDebInfo-76821e378ebd7f961b28.json","target-RendererBin-Debug-292f26fe5c04a3b25179.json","target-RendererBin-MinSizeRel-03c5e865ccaf3a320bd3.json","target-RendererBin-Release-f16679731f1179e70256.json","target-RendererBin-RelWithDebInfo-00f89c4a83ceebb09b8f.json","target-RendererLib-Debug-d86ecaa475f2f0eeb781.json","target-RendererLib-MinSizeRel-f03c28a900e77c8fadb6.json","target-RendererLib-Release-1f3d45e4ca4c0960764c.json","target-RendererLib-RelWithDebInfo-2d0c4faa02b8ab246bda.json","target-ZERO_CHECK-Debug-66bbffd0b2e5f79a4693.json","target-ZERO_CHECK-MinSizeRel-66bbffd0b2e5f79a4693.json","target-ZERO_CHECK-Release-66bbffd0b2e5f79a4693.json","target-ZERO_CHECK-RelWithDebInfo-66bbffd0b2e5f79a4693.json"]
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:35.124Z [debug] [cache] Reading CMake cache file g:/Code/CMake based projects/Mania Replay Renderer/build/CMakeCache.txt
console.ts:137 [Extension Host] [CMakeTools] 2020-09-10T11:32:35.127Z [debug] [cache] Parsing CMake cache string
console.ts:137 [Extension Host] provideCustomConfiguration
```

### Platform and Versions

- **Operating System**: Windows 10 2004
- **CMake Version**: 3.17.3
- **VSCode Version**: 1.48.2
- **CMake Tools Extension Version**: 1.4.1
- **Compiler/Toolchain**: Visual Studio 2019 with vcpkg toolchain file

### Other Notes/Information

Search dir list as reported by C/C++ extension:
```
include: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\BUILDTOOLS\VC\TOOLS\MSVC\14.27.29110\INCLUDE
include: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\BUILDTOOLS\VC\TOOLS\MSVC\14.27.29110\ATLMFC\INCLUDE
include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\UM
include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\UCRT
include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\SHARED
include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\WINRT
include: C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\CPPWINRT
```

**Comment by andreeis** [comment]
Thank you for reporting this issue. Can you share with us a small repro for this problem?

**Comment by longnguyen2004** [comment]
Seems like it has been fixed either by CMake Tools 1.4.1 or CMake itself, anyhow I'll close this now, thanks for your support!
